sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionBuilder.cs

```diff
@@ -80,192 +82,242 @@ public Instruction Create(SyntaxNode syntaxNode)
+            return new[]
+            {
+                CreateInstruction(objectCreationExpression, UcfgMethod.Assignment, CreateTempVariable(), ConstantExpression),
+                AddMethodCall(objectCreationExpression.Type, ctorSymbol, BuildArguments())
+            };
+
+            Expression[] BuildArguments()
+            {
+                // When building the args of the method call we need to pass the instance creation as first argument.
+                var methodCallArgs = new List<Expression> { GetMappedExpression(objectCreationExpression) };
+
+                if (objectCreationExpression.ArgumentList != null)
+                {
+                    methodCallArgs.AddRange(
+                        objectCreationExpression.ArgumentList.Arguments.Select(a => a.Expression).Select(GetMappedExpression));
+                }
 
-            return AddMethodCall(objectCreationExpression, ctorSymbol, arguments);
+                return methodCallArgs.ToArray();
+            }
         }
 
-        private Instruction CreateFromIdentifierName(IdentifierNameSyntax identifierName)
+        private IEnumerable<Instruction> CreateFromIdentifierName(IdentifierNameSyntax identifierName)
         {
             var identifierSymbol = GetSymbol(identifierName);
 
             if (identifierSymbol is IPropertySymbol property)
             {
-                return AddMethodCall(identifierName, property.GetMethod);
+                yield return AddMethodCall(identifierName, property.GetMethod, BuildPropertyGetterArguments().ToArray());
             }
-            else if (IsLocalVarOrParameterOfTypeString(identifierSymbol))
+            else if (IsLocalVarOrParameter(identifierSymbol))
             {
-                return CreateVariable(identifierName, identifierName.Identifier.Text);
+                CreateVariable(identifierName, identifierName.Identifier.Text);
             }
             else
             {
-                return CreateConstant(identifierName);
+                CreateConstant(identifierName);
             }
-        }
 
-        private Instruction CreateFromVariableDeclarator(VariableDeclaratorSyntax variableDeclarator)
-        {
-            if (variableDeclarator.Initializer != null)
+            IEnumerable<Expression> BuildPropertyGetterArguments()
             {
-                var variable = semanticModel.GetDeclaredSymbol(variableDeclarator);
-                if (IsLocalVarOrParameterOfTypeString(variable))
+                if (property.IsStatic)
+                {
+                    yield return CreateStaticCallExpression(property.ContainingType);
+                }
+                else
                 {
-                    return CreateAssignment(
-                        variableDeclarator,
-                        variable.Name,
-                        variableDeclarator.Initializer.Value);
+                    yield return ThisExpression;
```

Comment:
what about "var x = other.Property"? isn't it going to return `%0 Property [this]`?